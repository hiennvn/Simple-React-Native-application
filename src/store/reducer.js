import { combineReducers } from 'redux'

import navigation from '@navigation/reducer'
import appStore from '@app/reducer'

import main from '@scenes/Main/reducer'
import splash from '@scenes/Splash/reducer'
import login from '@scenes/Login/reducer'

export default combineReducers({
  navigation,
  appStore,
  main,
  splash,
  login,
})
