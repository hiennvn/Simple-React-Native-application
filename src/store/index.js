import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'
import reducer from './reducer'

export default preloadedState =>
  createStore(reducer, preloadedState, applyMiddleware(thunk, createLogger()))
