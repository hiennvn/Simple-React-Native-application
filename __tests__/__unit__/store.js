import store from '@store'
// import reducer from '@store/reducer'

describe('Store', () => {
  it('creates correctly', () => {
    expect(store()).toMatchSnapshot()
  })
})
