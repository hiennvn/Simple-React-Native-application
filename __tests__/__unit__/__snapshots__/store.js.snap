// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store creates correctly 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
